state/subnets.go

```diff
@@ -219,26 +220,26 @@ func (s *Subnet) Update(args SubnetInfo) error {
 		if err != nil {
 			return nil, err
 		}
-		var set bson.D
+		var bsonSet bson.D
 		if makeSpaceNameUpdate {
-			set = append(set, bson.DocElem{"space-name", args.SpaceName})
+			bsonSet = append(bsonSet, bson.DocElem{"space-name", args.SpaceName})
 		}
-		// TODO (hml) 2019-07-16
-		// Updated once SubnetInfo.AvailablityZone is a slice
-		if s.doc.AvailabilityZone == "" && args.AvailabilityZone != "" {
-			set = append(set, bson.DocElem{"availabilityzone", args.AvailabilityZone})
+		if len(args.AvailabilityZones) > 1 {
+			newAZ := set.NewStrings(args.AvailabilityZones...)
+			newAZ = newAZ.Difference(set.NewStrings(s.doc.AvailabilityZones...))
```

Comment:
Although I can't think of this being recruited for any case other than when there are no AZs and then the subnet info is updated with 1 or more, I think it is worth being explicit about it - treat it like the others and only set it if `len(s.doc.AvailabilityZones) == 0`.

Reply:
Since AZs are a slice, for update, I wanted to add any new AZ from the provider.  In the other cases we don't want to change a set value.